web/modules/custom/collection/collection.module

```diff
@@ -81,24 +104,75 @@ function collection_entity_insert(EntityInterface $entity) {
 }
 
 /**
+ * Implements hook_ENTITY_TYPE_access().
+ *
+ * Interrupts the permissions for updating a collection. The permission is not
+ * granted if there are no transitions allowed for the current state of the
+ * node.
+ */
+function collection_rdf_entity_access(EntityInterface $entity, $operation, AccountInterface $account) {
+  if (\Drupal::currentUser()->hasPermission('bypass node access')) {
+    return AccessResult::neutral();
+  }
+  if ($entity->bundle() != 'collection') {
+    return AccessResult::neutral();
+  }
+
+  if ($operation == 'edit') {
+    $allowed_transitions = $entity->field_ar_state->first()->getTransitions();
+    if (empty($allowed_transitions)) {
+      return AccessResult::forbidden();
+    }
+  }
+
+  return AccessResult::neutral();
+}
+
+/**
+ * Implements hook_ENTITY_TYPE_presave().
+ *
+ * Default graph is published. Draft graph is draft.
+ */
+function collection_rdf_entity_presave(EntityInterface $entity) {
+  $enabled_bundles = \Drupal::config('rdf_draft.settings')->get('revision_bundle_rdf_entity');
+  if (empty($enabled_bundles) || $entity->bundle() != 'collection' || !in_array($entity->bundle(), $enabled_bundles)) {
+    return;
+  }
+
+  $state = $entity->field_ar_state->value;
+  if (in_array($state, ['draft', 'proposed'])) {
+    \Drupal::entityManager()->getStorage($entity->getEntityTypeId())->getGraphHandler()->setTargetGraphToEntity($entity, 'draft');
+  }
+  elseif (in_array($state, [
+    'validated',
+    'archival_request',
+    'deletion_request',
+    'archived',
+  ])) {
+    \Drupal::entityManager()->getStorage($entity->getEntityTypeId())->getGraphHandler()->setTargetGraphToEntity($entity, 'default');
+  }
+  elseif (empty($state)) {
+    throw new \Exception("State field not found while trying to get the appropriate graph.");
+  }
+  else {
+    throw new \Exception("State field value is invalid. Current value found: " . $state);
+  }
+}
+
+/**
  * Implements hook_entity_update().
  *
- * Invalidate the og_content cache tag of the groups this entity belongs to.
+ * When the entity is saved into the default graph, the draft must be deleted.
  */
 function collection_entity_update(EntityInterface $entity) {
-  // @todo Fix cache invalidating.
-  // @see ISAICP-2475.
-  if (TRUE) {
+  $enabled_bundles = \Drupal::config('rdf_draft.settings')->get('revision_bundle_rdf_entity');
+  if (empty($enabled_bundles) || $entity->bundle() != 'collection' || !in_array($entity->bundle(), $enabled_bundles)) {
     return;
   }
-  $entity_type_id = $entity->getEntityTypeId();
-  $bundle = $entity->bundle();
-  if (Og::isGroupContent($entity_type_id, $bundle)) {
-    foreach (Og::getGroupIds($entity) as $entity_type => $ids) {
-      foreach ($ids as $id) {
-        Cache::invalidateTags(['og_group_content:' . $id]);
-      }
-    }
+
+  $state = $entity->field_ar_state->value;
+  if (in_array($state, ['validated'])) {
```

Comment:
I will fix this that way but later on, when the merge actually happens, this will also need to be generic and the generic rule implies that more than one states can exist in that array.